app/models/katello/upstream_pool.rb

```diff
@@ -13,13 +13,20 @@ def fetch_pools(params)
           included_fields: included_field_params(quantities_only)
         )
 
-        upstream_response = CP_POOL.get(params: cp_params)
+        upstream_response = fetch_upstream_with_error_handling(cp_params)
         pools = response_to_pools(upstream_response, pool_id_map: pool_id_map)
         total = upstream_response.headers[total_count_header] || pools.count
 
         respond(pools, total)
       end
 
+      def fetch_upstream_with_error_handling(cp_params)
+        CP_POOL.get(params: cp_params)
+      rescue RestClient::Gone
+        raise "The Subscription Allocation providing the imported manifest has been removed. " \
```

Comment:
Error looks good!

I have one suggestion around the error handling w.r.t `RestClient::Gone` which is I don't think that we should have HTTP / HTTP client lib errors being handled in our models.

We'll probably move away from it one day and the smaller the affected area the better it will be. One straightforward solution is to rescue `RestClient::Gone` in the resource class `Resources::Candlepin::UpstreamPool` and raise a new custom error, something like Katello::Errors::UpstreamConsumerGone and rescue from it in this code. What do you think?